lib/puppet/file_serving/fileset.rb

```diff
@@ -40,6 +40,7 @@ def initialize(path, options = {})
     self.links = :manage
     @recurse = false
     @recurselimit = :infinite
+    @max_files = 1000 # FIXME
```

Comment:
I was thinking the logic would be:

If `max_files` is not explicitly set (default) and the number of files exceeds some hard coded value (1000 seems fine), then issue a warning, but continue applying the resource(s) as it does now.
If `max_files` is set and it is exceeded, then error.

Reply:
Yes, I think this is a better solution to be delivered in a `y` release